js/foam/grammars/CSSRewriteURL.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * @license
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+CLASS({
+  package: 'foam.grammars',
+  name: 'CSSRewriteURL',
+
+  requires: [
+    'XHR',
+    'foam.grammars.CSSDecl',
+    'foam.util.Base64Encoder',
+  ],
+  imports: [
+    'assert',
+    'document',
+  ],
+
+  documentation: function() {/* A slightly modified
+    $$DOC{ref:'foam.grammars.CSSDecl'} prefix-rewrite parser that also knows how
+    to expand remote URLs to dataURLs after fetching the data.
+  */},
+
+  constants: {
+    PREFIXES: [
+      '-webkit-'
+    ],
+    PREFIXED_KEYS: {
+      'align-content': true,
+      'align-items': true,
+      'align-self': true,
+      'animation': true,
+      'box-shadow': true,
+      'column-count': true,
+      'column-gap': true,
+      'column-rule': true,
+      'display': 'flex',
+      'filter': true,
+      'flex': true,
+      'flex-basis': true,
+      'flex-direction': true,
+      'flex-flow': true,
+      'flex-grow': true,
+      'flex-shrink': true,
+      'flex-wrap': true,
+      'font-feature-settings': true,
+      'hyphens': true,
+      'justify-content': true,
+      'keyframes': true,
+      'order': true,
+      'transform': true,
+      'transform-origin': true,
+      'user-select': true
+    },
+  },
+
+  properties: [
+    {
+      name: 'urls',
+      lazyFactory: function() { return {}; },
+    },
+    {
+      name: 'declParser',
+      lazyFactory: function() { return this.CSSDecl.create(); },
+    },
+    {
+      name: 'parser',
+      lazyFactory: function() {
+        return SkipGrammar.create(this.parser_, seq('/*', repeat(not('*/', anyChar)), '*/'));
+      },
+    },
+    {
+      name: 'parser_',
+      lazyFactory: function() {
+        var css = this;
+        var r = function() { return str(repeat.apply(this, arguments)); };
```

Comment:
I think it's simpler to just use str(repeat(...)) or str(plus(...)) than for someone to have to figure out what you're doing here and then later in the grammar.